components/tabs/tab-mixin.js

```diff
@@ -135,7 +141,21 @@ export const TabMixin = superclass => class extends SkeletonMixin(superclass) {
 	}
 
 	#handleClick() {
+		this._clicked = true;
+		const beforeSelectedEvent = new CustomEvent('d2l-tab-before-selected', {
+			detail: {
+				select: () => { this.selected = true; this._clicked = false; },
+				reset: () => this._clicked = false
```

Comment:
A consumer would either run `e.detail.select` when they are ready to proceed with selection, or `e.detail.reset` if they do not wish to select in order to remove the hover styles. Let me know thoughts on this; we could also have it be just one function, e.g.:
```
update: (newSelection) => { this.selected = newSelection; this._clicked = false; }
```

Reply:
I'm ok with either.